crates/sui-framework/packages/deepbook/tests/clob_tests.move

```diff
@@ -4090,6 +4107,540 @@ module deepbook::clob_test {
         end(test)
     }
 
```

Comment:
Are these tests trying to skip self matching, or test for self matching? The test name is a little confusing

Reply:
These tests cover all cases in which self-matching occurs, and we are attempting to verify that we correctly cancel the oldest order and refund the user.
"with_skipping_self_matching" means that when self-matching occurs, we cancel the oldest order, effectively skipping the self-matching process by canceling the oldest order.

Reply:
And for "self_matching_prevention", we now only want the cancel_oldest behavior, so I will delete other options.

Reply:
Yes, I will add comments to explain the test scenario.